Fix misidentified init system

jsf9k opened the pull request:
* Add code to `prepare.yml` to install init on Debian images.  This is a workaround for the fact that testinfra currently misidentifies the init system on these hosts.  See philpep/testinfra#416 for more details.
* I also took the opportunity to pull in upstream changes.
* During the time that this PR sat unmerged, @felddy moved the skeleton projects from TravisCI to Github Actions.  I went ahead and made that transition here as well, and it wasn't entirely trivial because of the requirement for AWS IAM credentials for molecule testing.  See commits b8ca34d, e49432d, and 3b6e92c for details.

jsf9k committed:
Add support for Debian Buster

Buster is scheduled to be released today.  Buster also includes the
freeipa-client package, while Stretch does not, so switching to Buster
would make our lives a lot easier going forward.

jsf9k committed:
Merge pull request #16 from cisagov/improvement/add_buster

Add support for Debian Buster

jsf9k committed:
Use a prerelease of molecule instead of the latest from master

The latest from master is currently broken, but the latest prerelease
has all the goodies we need.  Once a new release is made we will use
that instead.

jsf9k committed:
Merge pull request #17 from cisagov/bugfix/upstream_molecule_broken

Use a prerelease of molecule instead of the latest from master

jsf9k committed:
Add updated Dockerfile.j2 and INSTALL.rst files

These skeleton files come from the latest release of molecule,
generated via the command "molecule init scenario".

jsf9k committed:
Add latest .yamllint generated via molecule init scenario

The generated file contains the truthy: disable bit that was in our
old file.

jsf9k committed:
Add fedora 29 and 30 to molecule configuration

jsf9k committed:
Trigger LGTM

jsf9k committed:
Merge pull request #18 from cisagov/improvement/updated_skeleton_files_from_latest_molecule

Add updated "molecule init scenario" files

jsf9k committed:
Add updates from running pre-commit autoupdate

jsf9k committed:
Merge pull request #17 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge pull request #19 from cisagov/improvement/upstream_updates

Upstream updates

jsf9k committed:
Change "company" name to the latest trend

The new name agrees with the naming guidance that was handed down by
Bobby Thompson.

jsf9k committed:
Merge pull request #20 from cisagov/change_company_name

Change "company" name to the latest trend

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into bugfix/fix-misidentified-init-system

jsf9k committed:
Add code to prepare.yml to install init on Debian images

This is a workaround for the fact that testinfra currently
misidentifies the init system on these hosts.

See philpep/testinfra#416 for more details.

jsf9k commented:
The fact that the (unneeded and unwanted) JavaScript LGTM analysis fails is a red herring. Please disregard.

dav3r reviewed (approved):
🏃

jsf9k commented:
Holster that gun, @dav3r!  I haven't resolved the failing TravisCI builds yet.

![image](https://user-images.githubusercontent.com/5521725/65965199-36c3b400-e42c-11e9-87c8-603bd6673b06.png)

dav3r commented:
Oops, I thought this was failing on the LGTM check- I didn't realize it was Travis failing here. 🤕

dav3r committed:
Add updates from running pre-commit autoupdate

dav3r committed:
Merge pull request #18 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

dav3r committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_in_upstream_changes

felddy committed:
Replace Travis-CI with GitHub actions.

felddy committed:
Fix action shield link

felddy committed:
Disable ansible-lint until they fix their setup.py
See: https://github.com/ansible/ansible-lint/issues/590

dav3r committed:
Merge pull request #21 from cisagov/improvement/merge_in_upstream_changes

Merge in upstream changes

felddy committed:
Remove extra line.

felddy committed:
Merge pull request #20 from cisagov/improvement/actions

Replace Travis-CI with GitHub actions

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/actions

felddy committed:
Add molecule run

felddy committed:
Remove white space that makes some of the team itchy.

felddy committed:
Merge pull request #22 from cisagov/improvement/actions

Replace Travis-CI with GitHub actions

mcdonnnj committed:
Re-enabled ansible-lint and ran pre-commit autoupgrade.

mcdonnnj committed:
Merge pull request #21 from cisagov/pre-commit_autoupdate_2019-11-02

Re-enable ansible-lint and autoupgrade.

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/upstream

felddy committed:
Remove vestigial travis file.

felddy committed:
Add GitHub action caching of pre-commit hooks and pip packages.

felddy committed:
Merge branch 'develop' into improvement/action_cache

felddy committed:
Remove search for previous pre-commit caches.
These seem to be very large caches.  Restoring an old one and updating 
it results in a cache larger than the maximum allowed cache size: 200MB. 
 
"Cache size of 254757924 bytes is over the 200MB limit, not saving 
cache."
So if the config changes it is best to just take the cache-miss and 
start from scratch.

felddy committed:
Merge pull request #23 from cisagov/improvement/upstream

Pull in upstream: pre-commit updates, and clean out travis

felddy committed:
Merge pull request #22 from cisagov/improvement/action_cache

Add GitHub action caching of pre-commit hooks and pip packages.

felddy committed:
Add GitHub Action caches for pip and pre-commit.

felddy committed:
Split build workflow into lint and test jobs.
This should speed things up a bit since pre-commit checks are slow, and 
so is the molecule run.

felddy committed:
Merge pull request #25 from cisagov/improvement/action_caches

Add GitHub Action caches for pip and pre-commit, split workflow into two jobs.

felddy committed:
Force pre-commit hooks to use python3. 
Fixes #23

felddy committed:
Autoupdate pre-commit hooks.

felddy committed:
Merge pull request #24 from cisagov/improvement/pre_commit_default_python

Set a default language for pre-commit hooks

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/upstream

felddy committed:
Merge pull request #26 from cisagov/improvement/upstream

Improvement/upstream

felddy committed:
Add codeowners file with team OIS maintainers.
See: 
https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

felddy committed:
Merge pull request #25 from cisagov/improvement/codeowners

Add codeowners file with team OIS maintainers.

felddy committed:
Update CODEOWNERS

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/codeowners

felddy committed:
Merge pull request #27 from cisagov/improvement/codeowners

Improvement/codeowners

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into bugfix/fix-misidentified-init-system

jsf9k committed:
Add foo so that AWS IAM secrets can be used by the test workflow

jsf9k committed:
Add boto3 as a testing requirement

When GitHub Actions tries to pull the GitHub OAUTH token from AWS SSM,
it needs boto3/botocore in order to grok the AWS IAM credentials it
gets from GitHub Secrets.

jsf9k committed:
Specify the AWS region as part of the IAM credentials

This is necessary for the AWS SSM call to succeed.

jsf9k commented:
Requesting re-review, since I wanted to make folks aware that the changes I made in b8ca34d, e49432d, and 3b6e92c will need to be made in any other Ansible role repos that require AWS IAM credentials for molecule testing.

It's also been a geological age since this PR was created, so it doesn't hurt to have folks take another look at it.

felddy commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,76 @@
+      - name: Run molecule tests
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
```
Nothing wrong with doing this... why did you make the default region a secret?

jsf9k replied:
@felddy No deep reason, other than keeping all those parameters together. If I did it in plain text I predicted @dav3r would suggest we were leaking  information. :wink:

I'm fine with doing it either way; I will adopt whatever policy the group decides.

felddy replied:
I was only asking to see if I should change the way I did it.  I can present some prior art of not putting `AWS_DEFAULT_REGION` in secrets:

https://github.com/cisagov/skeleton-packer/blob/develop/.github/workflows/release.yml
https://github.com/cisagov/skeleton-packer/blob/develop/.github/workflows/prerelease.yml

AMI creation is a more complicated situation where the regions usage is variable and knowing the value is important to end product.   Not saying change it.  Just saying.

jsf9k replied:
What do you think, @dav3r and @mcdonnnj?

mcdonnnj replied:
I'm fine either way as I see no issues with having it in the clear. However, I do think it's better if we pick which way and do it consistently.

felddy replied:
Until now it has been in the clear in the Travis configs:
https://github.com/cisagov/ansible-role-cyhy-runner/blob/dd5cecce629272c882d004c75a51b96ebb9518e1/.travis.yml#L10

A couple of things to note about how secrets work with GitHub Actions.  When you declare a secret, and access it in an action, the value of the secret is added to an output mask.  Anywhere the secret appears will be replaced with a `***`.  This is good.  

Until you use a common value as your secret.  I have a debug ssh step in the packer project that is toggled by the existence of a secret.  I used to have it check for the value `true`, but then everywhere true was output was replaced with `***`.  So I switched it to just a definition check so that I can just but gibberish in there and trip the toggle, while still seeing both boolean values clearly in the logs.  

https://github.com/cisagov/skeleton-packer/blob/d9264d11a2c13a318c37100fed65f7c5096a9ce7/.github/workflows/build.yml#L87-L89

Anyway, if the default region is defined as a secret, then it will be masked everywhere as well.  This could be problematic if we want to jump onto the AWS Console and hunt down the instance where the test is running.  Since secrets can't be recovered from the GUI, and the value is masked in all the log output you'll have to guess where it is running.

Seeing that I just typed all this, it's evident that I changed my mind from 🤷‍♂ to 🙅‍♂.

felddy reviewed (approved):
Looks good!  Thanks for the re-review chance.  It has been many moons!
🌚🌑🌔🌕🌝

mcdonnnj reviewed (approved):
Looking good 🏎, but I do agree we should resolve how we store the AWS default region.

felddy commented on .travis.yml:
```diff
@@ -1,58 +0,0 @@
-
-env:
-  global:
-    - AWS_DEFAULT_REGION="us-east-1"
```
AWS_DEFAULT_REGION was in the clear with TravisCI.

jsf9k committed:
Convert AWS region from a secret to a plain-text value

This was determined by the OIS collective to be The Way Forward.
